### PR TITLE
Adding DirectComposition to Silk

### PR DESCRIPTION
**Pull request by curin**
Adding Microsoft's DirectComposition Library to Silk as another supported library.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/Silk.NET?pullRequest=986) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/Silk.NET?pullRequest=986) before we can accept your contribution.<br/><br/>:x: curin [sign now](https://cla.dotnetfoundation.org/dotnet/Silk.NET?pullRequest=986)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/Silk.NET?pullRequest=986) it.</sub>

**Comment by Perksey** [comment]
Is this still active?

**Comment by curin** [comment]
Direct Composition requires Direct2D, so this got put on hold until Direct2D was merged.

**Comment by Perksey** [comment]
Ah yeah right, forgot about that.

**Comment by Beyley** [comment]
@curin Pinging to let you know Direct2D has been merged, so if you would like to go forward with this PR, you should be able to rebase and force push

**Comment by curin** [comment]
Thanks for the reminder, I might have time to get to it today.

**Comment by curin** [comment]
This should be all set now

**Comment by Beyley** [comment]
Could you rebase and attempt to regenerate again? there seems to be some missing symbols and its causing CI to fail, also if possible (aka if its a reasonable amount of code, dw about it if itd take a few thousand lines of code and requires something like D3D to work), could you write a small example app in the src/Lab/ folder? I'm not well versed enough in any of the DirectX things aside from Direct3D to be able to write my own example for this

**Comment by curin** [comment]
I'll rebase and attempt to regenerate, see if I can fix the CI issues. 
As for an example, DirectComposition is used for compositing multiple render targets and interacting with Windows (for some indirect drawing techniques). Unfortunately, that means it requires either d3d or d2d to build on top of. I do want to get an example running, but I don't know if I'll be able to at the moment. If I get the time this weekend, I'll attach it to this PR, otherwise I don't want to push this out necessarily.

**Comment by Beyley** [comment]
> I'll rebase and attempt to regenerate, see if I can fix the CI issues. As for an example, DirectComposition is used for compositing multiple render targets and interacting with Windows (for some indirect drawing techniques). Unfortunately, that means it requires either d3d or d2d to build on top of. I do want to get an example running, but I don't know if I'll be able to at the moment. If I get the time this weekend, I'll attach it to this PR, otherwise I don't want to push this out necessarily.

Understandable if it takes time, feel free to delay it to *after* this PR is merged, a second PR adding an example if nessesary would be good.

**Review by Perksey — APPROVED** [review]
Thank you again for your amazing contribution, I have no objections with this. @Beyley any comments or shall we give this the clickity click?

File: generator.json
Comment (Beyley) [inline generator.json:1809]: Can you add the typemap native and no obsete enum control descriptors? the first has been retroactively applied on most Direct* ones and the second is applied to new ones, if typemap-native causes problems though, it can be ignored, aside from that it looks good

File: src/Microsoft/Silk.NET.DirectComposition/Structs/FRAMESTATISTICS.gen.cs
Comment (Beyley) [inline src/Microsoft/Silk.NET.DirectComposition/Structs/FRAMESTATISTICS.gen.cs:21]: @Perksey do you know what happened here? this is very odd...

File: src/Microsoft/Silk.NET.DirectComposition/Structs/TagCOMPOSITIONSTATS.gen.cs
Comment (Beyley) [inline src/Microsoft/Silk.NET.DirectComposition/Structs/TagCOMPOSITIONSTATS.gen.cs:20]: Here too, the name of the struct is wrong, i think something in our rename logic is bad, or theres a rouge typemap

**Comment by Beyley** [comment]
I dont think what i brought up is a hard blocker, but the naming issues are a bit concerning

File: src/Microsoft/Silk.NET.DirectComposition/Structs/FRAMESTATISTICS.gen.cs
Comment (Perksey) [reply src/Microsoft/Silk.NET.DirectComposition/Structs/FRAMESTATISTICS.gen.cs:21]: There are a few cases like this and I don't know why they happen, really we need to add another path in the namer to detect when the "friendly name" doesn't look friendly (i.e. is all caps). I think Humanizer can convert nicely from SCREAMING_SNAKE_CASE to PascalCase, but obviously at the moment we don't actually tell Humanizer what source case we have (so it has to make its best guess, which seems to be wrong)

File: src/Microsoft/Silk.NET.DirectComposition/Structs/TagCOMPOSITIONSTATS.gen.cs
Comment (Perksey) [reply src/Microsoft/Silk.NET.DirectComposition/Structs/TagCOMPOSITIONSTATS.gen.cs:20]: I personally would manually rename this one as I haven't added anything to remove the `tag` prefix, so I've manually renamed structs like this elsewhere anyway.

**Comment by Perksey** [comment]
@Beyley can you approve this please

**Review by Beyley — APPROVED** [review]
since build passes, yea looks good to me

**Pull request merged**
